python/src/aac/lang/language_context.py

```diff
@@ -0,0 +1,182 @@
+    plugins, 3rd party definitions (libraries), and user-defined definitions.
+
+    Attributes:
+        definitions: The list of all definitions currently in the LanguageContext
+    """
+
+    def __attrs_post_init__(self):
+        """Post init hook for attrs classes."""
+        self.definitions_name_dictionary = {definition.name: definition for definition in self.definitions}
+
+    definitions: list[Definition] = attrib(default=Factory(list), validator=validators.instance_of(list))
+
+    # Private attribute - don't reference outside the class.
+    definitions_name_dictionary: dict[str, Definition] = attrib(
+        init=False, default=Factory(dict), validator=validators.instance_of(dict)
+    )
+
+    def add_definition_to_context(self, definition: Definition):
+        """
+        Add the Definition to the list of definitions in the LanguageContext.
+
+        Args:
+            definition: The Definition to add to the context.
+        """
+        new_definition = copy.deepcopy(definition)
+
+        if new_definition.name not in self.definitions_name_dictionary:
+            self.definitions_name_dictionary[new_definition.name] = new_definition
+            self.definitions.append(new_definition)
+        else:
+            logging.debug(
+                f"Did not add definition '{new_definition.name}' to the context because one already exists with the same name."
+            )
+
+        if definition.is_extension():
+            self._apply_extension_to_context(definition)
+
+    def add_definitions_to_context(self, definitions: list[Definition]):
+        """
+        Add the list of Definitions to the list of definitions in the LanguageContext, any extensions are added last.
+
+        Args:
+            definitions: The list of Definitions to add to the context.
+        """
+
+        extension_definitions = get_definitions_by_root_key("ext", definitions)
+
+        # Avoiding the use of sorted() since we already deepcopy each definition as we add it to the context.
+        for definition in definitions:
+            if definition not in extension_definitions:
+                self.add_definition_to_context(definition)
+
+        for extension_definitions in extension_definitions:
+            self.add_definition_to_context(extension_definitions)
+
+    def get_root_keys(self) -> list[str]:
+        """
+        Get the list of root keys as defined in the LanguageContext.
+
+        Returns a list of strings indicating the current root keys in the active context. These
+        keys may differ from those provided by the core spec since the LanguageContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+        See :py:func:`aac.spec.get_root_keys()` for the list of root keys provided by the unaltered core AaC DSL.
+
+        Returns:
+            A list of strings, one entry for each root key in the LanguageContext.
```

Comment:
```suggestion
        Returns a list of strings indicating the current root keys in the active context. 
        
        Returns:
            A list of strings, one entry for each root key in the LanguageContext.
            
            These keys may differ from those provided by the core spec since the LanguageContext applies definitions
            from active plugins and user files, which may extend the set of root keys.
            See :py:func:`aac.spec.get_root_keys()` for the list of root keys provided by the unaltered core AaC DSL.

```